### ISSUE TITLE
form.post produces Mixed Content: The page at 'domain' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint

### ISSUE THREAD

**Issue by Arturexe** [inertiajs/inertia-laravel#416]
I've been trying to resolve an error that happens only in production. When I try to `create` a new database entry, following errors are thrown:

```
Mixed Content: The page at 'https://strong-moebel.art/admin/gallerie/neu' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://strong-moebel.art/admin/gallerie'. This request has been blocked; the content must be served over HTTPS.
```

```
Uncaught (in promise) Error: Network Error
    at wh (main.750d1ea1.js:4:96980)
    at s.onerror (main.750d1ea1.js:5:1837)
```

everything else works, including the `edit` method of the entries. I'm using a `resource controller`. `create` method uses inertia's `form.post` and `edit` method uses its `form.put`.

I've been trying to debug this with the solutions provided in:

1. [Mixed content issue- Content must be served as HTTPS](https://laracasts.com/discuss/channels/laravel/mixed-content-issue-content-must-be-served-as-https)
2. [Mixed Content (laravel)](https://stackoverflow.com/questions/47018712/mixed-content-laravel)

basically people are saying to add:
```
if (App::environment('production')) {
    URL::forceScheme('https');
}
```

to the `boot()` method of your `AppServiceProvider.php`. I've done so but the error still occurs.

On the server, I've tried:

```
APP_ENV=production
APP_URL=http://localhost
APP_URL=https://localhost
APP_URL=
APP_URL=http://strong-moebel.art
APP_URL=https://strong-moebel.art
```

but nothing seems to solve the issue. My webhost is cloudways and I'm using their `Let's Encrypt SSL Certificate`. I've also tried to remove the certificate and see what happens but even then the exact same error occurs. 

I'm using `vite` to build production built.

**Comment by chapdel** [comment]
Same issue here

**Comment by jonathan-brown** [comment]
Did you resolve this? I'm getting the exact same thing!

**Comment by Arturexe** [comment]
@jonathan-brown Hey, yes. Check out this [stack post](https://stackoverflow.com/questions/72590500/mixed-content-the-page-at-domain-was-loaded-over-https-but-requested-an-inse)

**Comment by jessarcher** [comment]
Hey there,

We're closing this issue because it's inactive, already solved, old or not relevant anymore. Feel free to open up a new issue if you're still experiencing this problem.

**Comment by larangin** [comment]
@Arturexe  thanks for the post. I solved it adding meta Tag as suggested: 
`<meta http-equiv="Content-Security-Policy" content="upgrade-insecure-requests">`

**Comment by Aigarsss** [comment]
Had the same issue, the suggested meta tag fixed it

**Comment by magboifo** [comment]
So i had the same error.

My build: Laravel 10, Inertia, Vue3.

It turns out that 'ziggy' generated a http url.

I found out by running ` console.log(this.route())`.

I had a https://example.com/,  ziggy generated http://example.com/.

So i deleted the ziggy.js file and regenerated it.

Result:
https://example.com/ all across the board. I hope this helps.

**Comment by alexmillman23** [comment]
Adding the metatag above actually broke my Inertia auth routes. Adding `URL::forceScheme('https');` to my AppServiceProvider boot method fixed it.

**Comment by jbiddulph** [comment]
was it easy enough to remove ziggy? Do I actually need Ziggy?
